source/server/server.cc

```diff
@@ -150,6 +151,12 @@ void InstanceImpl::flushStats() {
     server_stats_->total_connections_.set(numConnections() + info.num_connections_);
     server_stats_->days_until_first_cert_expiring_.set(
         sslContextManager().daysUntilFirstCertExpires());
+
+    Envoy::MutexData mutex_data = Envoy::MutexTracer::GetTracer()->GetData();
+    server_stats_->mutex_contention_count_.set(mutex_data.num_contentions);
```

Comment:
I don't have a strong opinion on whether these are treated as regular "server.*" stats vs. as a separate `/contention` admin page. @mattklein123 (since you assigned yourself to review) WDYT?

Reply:
To summarize why not stats:

 * Use relaxed lock
 * Atomics have process lifetime to match tracer registration, whereas stats have server lifetime, and those are different in unit tests.

Reasons to use stats:
 * Easier hook into sinks for monitoring.

I am ok either way and would go with whatever Matt says. If we do use stats we should be careful about reconnecting the new stats to the existing MutexTracer when the server reinitializes.

Reply:
To summarize why not stats:

 * Use relaxed lock
 * Atomics have process lifetime to match tracer registration, whereas stats have server lifetime, and those are different in unit tests.

Reasons to use stats:
 * Easier hook into sinks for monitoring.

I am ok either way and would go with whatever Matt says. If we do use stats we should be careful about reconnecting the new stats to the existing MutexTracer when the server reinitializes.